cdap-support-bundle/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -105,4 +139,130 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws NotFoundException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundles/{uuid}/status")
+  public void getSupportBundleByUUID(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.GET);
+    List<SupportBundleOperationStatus> bundleOperationStatusList = new ArrayList<>();
+    SupportBundleOperationStatus bundleOperationStatus = bundleGenerator.getSingleBundleJson(uuid);
+    bundleOperationStatusList.add(bundleOperationStatus);
+    responder.sendString(HttpResponseStatus.OK, GSON.toJson(bundleOperationStatusList));
+  }
+
+  /**
+   * Delete the specific bundle by using the uuid
+   *
+   * @param uuid the bundle id which is also the uuid
+   * @throws NotFoundException is thrown if status.json is not found
+   */
+  @DELETE
+  @Path("/support/bundles/{uuid}")
+  public void deleteSupportBundle(HttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.DELETE);
+    File baseDirectory = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR));
+    File uuidFile = new File(baseDirectory, uuid);
+    if (!baseDirectory.exists()) {
+      throw new NotFoundException("No content in Support Bundle.");
+    }
+    if (!uuidFile.exists()) {
+      throw new NotFoundException(String.format("No such uuid '%s' in Support Bundle.", uuid));
+    }
+    bundleGenerator.deleteOldFoldersIfExceedLimit(uuidFile);
+    responder.sendString(HttpResponseStatus.OK, String.format("Successfully deleted bundle %s", uuid));
+  }
+
+  /**
+   * Exports all selected files as a ZIP archive file.
+   *
+   * @param uuid the bundle id which is also the uuid fileJson body
+   */
+  @POST
+  @Path("/support/bundle/{uuid}/download")
+  public void appsExport(FullHttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.GET);
+    String requestContent = request.content().toString(StandardCharsets.UTF_8);
+    if (requestContent == null || requestContent.length() == 0) {
+      throw new BadRequestException("Request body is empty.");
+    }
+    SupportBundleExportRequest bundleExportRequest =
+      GSON.fromJson(requestContent, SupportBundleExportRequest.class);
+    File tempDir = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR),
+                            cConf.get(Constants.SupportBundle.SUPPORT_BUNDLE_TEMP_DIR)).getAbsoluteFile();
```

Comment:
You should just use `SUPPORT_BUNDLE_TEMP_DIR` config directly instead of having it inside the local data dir. In the `cdap-default.xml`, have the value set to `${local.data.dir}/tmp`.